system/smbus.c

```diff
@@ -1348,6 +1350,24 @@ static bool ich5_get_smb(void)
     return (smbusbase != 0);
 }
 
+// ----------------------
+// VIA SMBUS Controller
+// ----------------------
+// detection based on Linux/i2c-viapro
+static bool viapro_get_smb(void)
```

Comment:
SMB base address needs to be passed from the caller, per my comment on the PR :)